src/components/member-profile/contribution-type/index.js

```diff
@@ -27,11 +27,25 @@ const renderActiveTasks = (tasks) => {
 
 const ContributionType = (props) => {
   const { fullName, type, imageLink, contributions, devUser, tasks } = props;
+  const [showMoreContent, setShowMoreContent] = useState(false);
 
-  const [showMoreContent, setShowMoreContent] = useState(true);
+  useEffect(() => {
+    if (type === 'Noteworthy' || type === 'All') {
+      setShowMoreContent(contributions.length > 0);
+      return;
+    }
+    setShowMoreContent(tasks.length > 0);
+  }, [tasks.length, contributions.length]);
 
   const showMoreContentHandler = () => {
-    setShowMoreContent((prevstate) => !prevstate);
+    if (type === 'Noteworthy' || type === 'All') {
+      if (contributions.length > 0) {
+        setShowMoreContent(!showMoreContent);
+      }
+    }
+    if (tasks.length > 0) {
+      setShowMoreContent(!showMoreContent);
```

Comment:
here as well ?
shouldnt it be either `prev=>!prev` or `false/true` based on whatever the situation is?

Reply:
this shouldn't be done but the other methods are not working as expected will need to refactor this file.